### DISCUSSION TITLE
Authenticate user with pem certificate in keycloak rest api

### DISCUSSION THREAD

**Discussion by pravinfullstack** [keycloak/keycloak#9373] [Q&A]
I am running keycloak using docker with an intention to authenticate user via rest api. Currently, my configuration includes using grant_type=password along with client-id and client-secret to authenticate a client.

```
curl --location --request POST 'http://localhost:8079/auth/realms/demo/protocol/openid-connect/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'client_id=admin-cli' \
--data-urlencode 'client_secret=234k23jk42j23i4u23iwerwlekrjwl'
--data-urlencode 'username=thor' \
--data-urlencode 'password=thor' \
--data-urlencode 'grant_type=password'

```
I have set to use client authenticator as signed jwt for my client for which I have a PEM file.

<img width="808" alt="image" src="https://user-images.githubusercontent.com/11708491/147871558-1515397b-8f48-4392-beed-02f6415e3141.png">

**How do I supply PEM file in REST API to get access_token for the user ?**

**Comment by gschaden** [comment]
Check documentation at: https://wjw465150.gitbooks.io/keycloak-documentation/content/index.html
